provenance.jq

```diff
@@ -0,0 +1,80 @@
+# input: "build" object (with "buildId" top level key)
+# output: array of image tags
+def tags:
+  .source.arches[].tags[],
+  .source.arches[].archTags[],
+  .build.img
+;
+
+# input: "build" object (with "buildId" top level key)
+# output: purl platform query string
+def platform_string:
+  .source.arches[].platformString | gsub("/"; "%2F")
+;
+
+# input: "tags" object with image digest and platform arguments
+# output: json object for in-toto provenance subject field
+def subjects($platform; $digest):
+  {
+      "name": ("pkg:docker/" + . + "?platform=" + $platform),
+      "digest": {
+        "sha256": $digest
+      }
+  }
+;
+
+# input: GITHUB context argument
+# output: json object for in-toto provenance external parameters field
+def github_external_parameters($context):
+($context.workflow_ref | gsub( $context.repository + "/"; "")) as $workflowPathRef |
+{
+  inputs: $context.event.inputs,
+  workflow: {
+    ref: ($workflowPathRef | split("@")[1]),
+    repository: ($context.server_url + "/" + $context.repository),
+    path: ($workflowPathRef | split("@")[0]),
+    digest: {sha256: $context.workflow_sha}
+  }
+}
+;
+
+# input: GITHUB context argument
+# output: json object for in-toto provenance internal parameters field
+def github_internal_parameters($context):
+{
+  github: {
+    event_name: $context.event_name,
+    repository_id: $context.repository_id,
+    repository_owner_id: $context.repository_owner_id,
+  }
+}
+;
```

Comment:
I moved `github_internal_parameters` to be embedded in the main function and kept `github_external_parameters` separate

Reply:
Yeah, given they're documents embedded inside well-named keys, I think it's roughly equal (preamble of `def foo:` vs preamble of `foo:`) with a slight preference for embedding because then it matches the upstream documentation / output document slightly better.